packages/rest/src/body-parsers/helper.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import * as debugModule from 'debug';
+import {HttpError} from 'http-errors';
+import {Request, Response, RequestBodyParserOptions} from '../types';
+
+import {OptionsJson, OptionsUrlencoded, OptionsText} from 'body-parser';
+const debug = debugModule('loopback:rest:body-parser');
+
+/**
+ * Get the content-type header value from the request
+ * @param req Http request
+ */
+export function getContentType(req: Request): string | undefined {
+  return req.get('content-type');
+}
+
+/**
+ * Express body parser function type
+ */
+export type BodyParserWithCallback = (
+  request: Request,
+  response: Response,
+  callback: (err: HttpError) => void,
+) => void;
+
+/**
+ * Normalize parsing errors as `4xx`
+ * @param err
+ */
+export function normalizeParsingError(err: HttpError) {
+  debug('Cannot parse request body %j', err);
+  if (!err.statusCode || err.statusCode >= 500) {
+    err.statusCode = 400;
+  }
+  return err;
+}
+
+// tslint:disable:no-any
+
+/**
+ * Parse the body asynchronously
+ * @param handle The express middleware handler
+ * @param request Http request
+ */
+export function parseRequest(
```

Comment:
I find this name misleading because the function is not doing any body parsing, just executing a body-parsing middleware in a way that's async/await friendly. Let's find a name that described this intent better. 

For example: `executeBodyParserMiddleware`.

I think this function can be very useful to extension authors building custom body parsers, because it effectively allows them to take any Express middleware performing body parsing and wrap it into LB4 RequestParser. It would be great to change it into a well-documented public API.